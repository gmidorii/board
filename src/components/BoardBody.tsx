import * as React from 'react'
import { Message } from '../model/Model'

export interface Props {
  messages: Message[]
}
export interface State {}

export class BoardBody extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
  }

  render() {
    const messages: JSX.Element[] = []
    this.props.messages.forEach(element => {
      messages.push(
        <div key={element.id} className="message">
          <div>
            <span>{element.user}</span>
            <span className="message-date">
              {element.createdAt.toLocaleString()}
            </span>
          </div>
          <div>{element.body}</div>
        </div>
      )
    })

    return messages
  }
}
