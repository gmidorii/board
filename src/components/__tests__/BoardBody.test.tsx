import * as React from 'react'
import * as render from 'react-test-renderer'
import { BoardBody } from '../BoardBody'

test('body snapshot', () => {
  const d = new Date('2018-09-16T15:00:00')
  const com = render.create(
    <BoardBody
      messages={[{ id: 'id', user: 'user', body: 'body', createdAt: d }]}
    />
  )
  expect(com.toJSON()).toMatchSnapshot()
})
