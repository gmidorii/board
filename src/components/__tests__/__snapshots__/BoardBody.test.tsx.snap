// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`body snapshot 1`] = `
<div
  className="message"
>
  <div>
    Date: 
    2018-9-16 15:00:00
  </div>
  <div>
    Body:
    <br />
    body
  </div>
</div>
`;
